openapi/src/main/scala/endpoints/algebra/DocumentedEndpoints.scala

```diff
@@ -0,0 +1,56 @@
+package endpoints
+package algebra
+
+import scala.language.higherKinds
+
+/**
+  * Algebra interface for describing endpoints including documentation
+  * (such as human readable descriptions of things).
+  *
+  * This interface is modeled after [[Endpoints]] but some methods
+  * take additional parameters carrying the documentation part.
+  */
+trait DocumentedEndpoints
+  extends DocumentedRequests
+    with DocumentedResponses {
```

Comment:
The diff is huge partly because all the algebra interfaces (e.g. `algebra.Endpoints`) are “mirrored” in “documented” algebra interfaces like this one.